crypto/merkle_test.go

```diff
@@ -95,3 +95,63 @@ func TestPaddedStorageProof(t *testing.T) {
 		t.Error("padded segment proof failed")
 	}
 }
+
+// TestReadSegments tests MerkleTree.ReadSegments by building a MerkleTree and
+// comparing its root with the root returned from ReaderMerkleRoot.
+func TestReadSegments(t *testing.T) {
+	// Generate leaf data.
+	numSegments := 7
+	data := make([]byte, numSegments*SegmentSize)
+	rand.Read(data)
```

Comment:
You should definitely add a test case where the data is not a multiple of `SegmentSize`.